### PR TITLE
feat(earn): Estimate market withdraw modal

### PR DESCRIPTION
**Pull request by th0rOdinson**

### CODE REVIEW

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx [hunk 1]
```diff
@@ -1,9 +1,132 @@
...
+    useEstimateMarketWithdraw({
+      strategy,
+      shouldRefetch: isOpen,
+      position,
+    });
+
+  return (
+    <ContainerBox flexDirection="column">
+      <ContainerBox gap={6}>
+        <CommonTransactionStepItem icon={<Wallet2Icon fontSize="large" />} isCurrentStep isLast={false} hideWalletLabel>
```
Comment (FiboApe) [inline apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx:82]: does this actually work ok and adds the line in the left to all of this?

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx [hunk 2]
```diff
@@ -1,9 +1,132 @@
...
+  intl: IntlShape;
+  isLoading?: boolean;
+  isNegative?: boolean;
+}) => (
+  <ContainerBox flexDirection="column" gap={1}>
+    <Typography variant="bodySmallRegular">{intl.formatMessage(title)}</Typography>
+    <ContainerBox gap={2} alignItems="center">
+      <TokenIcon token={token} size={6} />
+      <Typography variant="bodyBold" color={({ palette }) => colors[palette.mode].typography.typo2} lineHeight={1}>
+        {isLoading ? (
```
Comment (FiboApe) [inline apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx:34]: this probably should not have the () around the USD price, its a great use case for TokenAmount :)

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/hooks/useEstimateMarketWithdraw.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+export default function useEstimateMarketWithdraw({
+  strategy,
+  shouldRefetch,
+  position,
+}: UseEstimateMarketWithdrawProps) {
+  const earnService = useEarnService();
+  const [estimatedMarketAmount, setEstimatedMarketAmount] = React.useState<AmountsOfToken | undefined>();
+  const [isLoading, setIsLoading] = React.useState(false);
```
Comment (FiboApe) [inline apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/hooks/useEstimateMarketWithdraw.tsx:22]: shouldn't we be using the lastest pattern of having { result, isLoading, error} here? It triggers less rerenders

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx (hunk 1 above)
Comment (th0rOdinson) [reply apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx:82]: Yep!
Screenshot 2024-11-06 at 3 31 29 PM

File: apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx (hunk 2 above)
Comment (th0rOdinson) [reply apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx:34]: Just followed the figma here 😅 , but i'll prioritize your PR so we can implement it here

**Review by FiboApe — APPROVED** [review]

**Pull request merged**